biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/Traverser.java

```diff
@@ -207,7 +207,9 @@ private void parseResource(Archive archive, String parent) throws Exception {
 		ResourceBuilder rb = new ResourceBuilder();
 
 		Version frameworkVersion = toFrameworkVersion(archive.revision.version.getOSGiVersion());
-		String bsn = archive.revision.program.toString();
+
+		// g:a:jar and g:a:jar:all are different artifacts respect classifier
+		String bsn = archive.getWithoutVersion();
```

Comment:
Actually, this seems to be what MavenBndRepository does:

https://github.com/bndtools/bnd/blob/400cd7557624ba4428f074a9b3e0023ec220108c/biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java#L328-L334

However, MavenBndRepository does not set a osgi.identity (which `addReserveIdentity` does). Instead it adds an information capability which the BridgeRepository understands. We should change BndPomRepository to match the MavenBndRepository behavior here.

Reply:
We should. Let both work same as much as possible.

Could i address this in an extra pr or would you have it in this?

Reply:
I added a new commit.
MavenBndRepository handles name and version seperate.
```
String name = archive.getWithoutVersion();  // groupId:artifactId
addInformationCapability(rb, name, version.getOSGiVersion(), repo.toString(),
```
while BndPomRepository holds the version in the name.
```
addInformationCapability(rb, archive.toString(), parent); //groupId:artifactId:version
```
because of this I had  to change some Archive name calculations.`Archive.valueOf(a + ":" + version);`


@bjhargrave This is what i can do with my very limited knowledge about the repository.  Hope this is the right direction.